### PR TITLE
Preprocessing embed

### PR DESCRIPTION
**Pull request by szaboildi**
This PR implements a function in `ml_logic/preprocessing.py` that embeds and pads X based on the global parameter `EMBEDDING` (in `params.py`)

### CODE REVIEW

File: polclassifier/ml_logic/preprocessing.py [hunk 1]
```diff
@@ -160,3 +161,31 @@ def save_processed_to_cache(
...
     y.to_csv(y_path, index=False)
     return
+
+
+def embed_and_pad(X, embedding, stop_words=stop_words):
+    X_embed = embed_sentences(X, embedding, stop_words)
+    
+    maxlen = max([len(x) for x in X_embed])
+    X_pad = pad_sequences(X_embed, dtype='float32', padding='post', maxlen=maxlen)
+
```
Comment (Seoman3) [inline polclassifier/ml_logic/preprocessing.py:177]: return X_pad is missing

File: polclassifier/ml_logic/preprocessing.py [hunk 2]
```diff
@@ -160,3 +161,31 @@ def save_processed_to_cache(
...
+            embedded_sentence.append(embedding[word])
+        
+    return np.array(embedded_sentence)
+
+# Function that converts a list of sentences into a list of matrices
+def embed_sentences(embedding, sentences):
+    embed = []
+    
+    for sentence in sentences:
+        embedded_sentence = embed_sentence_with_TF(embedding, sentence.split())
```
Comment (Seoman3) [inline polclassifier/ml_logic/preprocessing.py]: Is the argument for stop_words necessary?

File: polclassifier/params.py [hunk 3]
```diff
@@ -32,7 +37,7 @@
 GAMMA_DEFAULT = "scale"
 C_DEFAULT = 4.2
 
+# Parameters for embedding
+EMBEDDING = "glove-wiki-gigaword-100" # code of embedding from gensim
```
Comment (Seoman3) [inline polclassifier/params.py:46]: In our notebooks we had glove-wiki-gigaword-50. Did you change it on purpose?

**Review by Seoman3 — COMMENTED** [review]
Had some minor comments/questions. I added them to the according lines.

File: polclassifier/ml_logic/preprocessing.py (hunk 1 above)
Comment (szaboildi) [reply polclassifier/ml_logic/preprocessing.py:177]: 🤦

File: polclassifier/ml_logic/preprocessing.py (hunk 2 above)
Comment (szaboildi) [reply polclassifier/ml_logic/preprocessing.py]: Thanks, fixed!

**Pull request merged**